File: app/src/main/java/com/example/thegreatmenu/FeedActivity.java
package com.example.thegreatmenu;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ListView;


import com.google.firebase.auth.FirebaseAuth;

import java.util.ArrayList;

public class FeedActivity extends BaseActivity {

    private ArrayList<String> allRecipes;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_feed);

        // נתונים לדוגמה
        allRecipes = new ArrayList<>();
        allRecipes.add("פסטה ברוטב עגבניות (חלבי)");
        allRecipes.add("חביתת ירק מושקעת (פרווה)");
        allRecipes.add("עוף בתנור עם תפוחי אדמה (בשרי)");
        allRecipes.add("סלט יווני עשיר (חלבי)");
        allRecipes.add("טוסט נקניק (בשרי)");
        allRecipes.add("שקשוקה חריפה (פרווה)");

        ListView listView = findViewById(R.id.listViewRecipes);
        Button btnAdd = findViewById(R.id.btnAddRecipe);
        Button btnLogout = findViewById(R.id.btnLogout);
        Button btnSettings = findViewById(R.id.btnSettings);
        // שימוש באדפטר המותאם אישית שלנו (RecipeAdapter)
        RecipeAdapter adapter = new RecipeAdapter(this, allRecipes);
        listView.setAdapter(adapter);

        // מעבר להוספת מתכון
        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(FeedActivity.this, AddRecipeActivity.class);
                startActivity(intent);
            }
        });

        // התנתקות
        btnLogout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FirebaseAuth.getInstance().signOut();
                Intent intent = new Intent(FeedActivity.this, MainActivity.class);
                // ניקוי ההיסטוריה
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                startActivity(intent);
                finish();
            }
        });

        btnSettings.setOnClickListener(v -> {
            startActivity(new Intent(FeedActivity.this, SettingsActivity.class));
        });


    }
}

File: app/src/main/java/com/example/thegreatmenu/FeedActivity.kt
package com.example.thegreatmenu

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.widget.addTextChangedListener
import com.google.firebase.auth.FirebaseAuth

class FeedActivity : AppCompatActivity() {

    val allRecipes = arrayListOf(
        "פסטה ברוטב עגבניות (חלבי)",
        "חביתת ירק מושקעת (פרווה)",
        "עוף בתנור עם תפוחי אדמה (בשרי)",
        "סלט יווני עשיר (חלבי)",
        "טוסט נקניק (בשרי)",
        "שקשוקה חריפה (פרווה)"
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_feed)

        val listView = findViewById<ListView>(R.id.listViewRecipes)
        val etSearch = findViewById<EditText>(R.id.etSearch)
        val btnAdd = findViewById<Button>(R.id.btnAddRecipe)
        val btnLogout = findViewById<Button>(R.id.btnLogout) // הכפתור החדש

        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, allRecipes)
        listView.adapter = adapter

        etSearch.addTextChangedListener { text ->
            adapter.filter.filter(text)
        }

        btnAdd.setOnClickListener {
            startActivity(Intent(this, AddRecipeActivity::class.java))
        }

        // לוגיקת התנתקות
        btnLogout.setOnClickListener {
            FirebaseAuth.getInstance().signOut() // התנתקות מ-Firebase

            // חזרה למסך הראשי
            val intent = Intent(this, MainActivity::class.java)
            // ניקוי ההיסטוריה כדי שלא יוכלו לחזור אחורה עם כפתור החזרה
            intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
            startActivity(intent)
            finish()
        }
    }
}